### PR TITLE
Use new survey link for Build Your Own workshops

### PR DESCRIPTION
**Pull request by TurnerRiley**
We are going to use a special post-workshop survey link for the new Build Your Own workshops. This PR updates where we obtain that survey link to return the new one for Build Your Own workshops, and still returns a Foorm otherwise.

This was tricky to test because the new survey page doesn't work locally since it is unfinished (there is a placeholder live now on production at [studio.code.org/form/professional_learning_educator_post_workshop_survey](https://studio.code.org/form/professional_learning_educator_post_workshop_survey) for teachers and [studio.code.org/form/professional_learning_facilitator_post_workshop_survey](https://studio.code.org/form/professional_learning_facilitator_post_workshop_survey) for facilitators), so I was making sure it sent to the right URL (which locally shows a sad bee). In addition, I don't have Foorm set up locally so I was testing that it sent to the proper URL and showed Foorm errors in the console. I felt comfortable not doing more to check Foorm since I didn't touch anything with Foorm itself and I didn't want to spend too much time on this ticket (but I'm open to pushback if this feels necessary).

### Post-workshop survey link on My PL page after workshop ends
Non-Build Your Own uses the foorm:
(I don't have Foorms set up locally but this shows that it's sending to the right url and has errors expecting values for Foorm)
right url

Build Your Own sends user to `/form/professional_learning_educator_post_workshop_survey`:
(this page is currently not ready so locally it shows a sad bee, but on production there is a placeholder at [/form/professional_learning_educator_post_workshop_survey](https://studio.code.org/form/professional_learning_educator_post_workshop_survey))
form link educator

#### `exit_survey` email
The survey is linked here in the email:
teacher survey email

For a non-BYO workshop, clicking that link sends the user to that page with the Foorm errors as expected:
teacher email non byo shows other

For a BYO workshop, clicking that link sends the user to the sad bee page with the right url:
teacher survey byo sad bee

#### `facilitator_post_workshop` email
The survey is linked here in the email:
facilitator email survey link

For a non-BYO workshop, clicking that link sends the user to that page with the Foorm errors as expected:
facilitator email sends to foorm non byo

For a BYO workshop, clicking that link sends the user to the sad bee page with the right url:
facilitator email sends to special byo

## Links
Jira ticket: [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?assignee=60d62161f65054006980bd71&selectedIssue=ACQ-2058)

## Testing story
Local testing and adding unit tests.

### CODE REVIEW

File: dashboard/app/controllers/pd/workshop_daily_survey_controller.rb [hunk 1]
```diff
@@ -64,16 +64,14 @@ def new_pre_foorm
     end
 
     # General post-workshop survey using foorm system.
-    # GET '/pd/workshop_post_survey?enrollmentCode=code'
```
Comment (bethanyaconnor) [inline dashboard/app/controllers/pd/workshop_daily_survey_controller.rb:67]: Why remove this?

File: dashboard/app/controllers/pd/workshop_daily_survey_controller.rb [hunk 2]
```diff
@@ -64,16 +64,14 @@ def new_pre_foorm
...
     # Enrollment code is an optional parameter, otherwise will show most recent workshop.
     #
     # If the post-survey has been already completed, will redirect to thanks page.
     def new_post_foorm
       new_general_foorm(survey_names: POST_SURVEY_CONFIG_PATHS, day: nil)
     end
 
-    def new_facilitator_post_foorm
```
Comment (bethanyaconnor) [inline dashboard/app/controllers/pd/workshop_daily_survey_controller.rb:75]: Is this a helper method and not a controller action?

File: dashboard/app/controllers/pd/workshop_daily_survey_controller.rb [hunk 3]
```diff
@@ -109,9 +107,23 @@ def new_general_foorm(survey_names:, day:, subject: FOORM_GENERAL_SURVEY_SUBJECT
...
-    # This will direct to new_post_foorm
+    # If Build Your Own, redirect to its specific survey link. Otherwise, use new_post_foorm.
     def new_post
-      return new_post_foorm
+      course = get_workshop_by_enrollment_or_course_and_subject(enrollment_code: params[:enrollment_code], course: nil, subject: nil)&.course
+      if course == COURSE_BUILD_YOUR_OWN
+        redirect_to CDO.studio_url SURVEY_LINKS[:COURSE_BUILD_YOUR_OWN_TEACHER], CDO.default_scheme
+      else
+        return new_post_foorm
+      end
```
Comment (bethanyaconnor) [inline dashboard/app/controllers/pd/workshop_daily_survey_controller.rb]: what's the difference between redirect and return here? ie why does one path use one and the other uses something else? Just want to wrap my ahead around it :)

**Review by bethanyaconnor — APPROVED** [review]
Some general questions but nothing blocking. Looks good!

File: dashboard/app/controllers/pd/workshop_daily_survey_controller.rb (hunk 1 above)
Comment (TurnerRiley) [reply dashboard/app/controllers/pd/workshop_daily_survey_controller.rb:67]: I removed this to try and simplify the logic a bit, plus this method was labeled with `foorm` which would be misleading for the new non-foorm survey. There's the method `new_post` which has a super similar controller action of `GET /pd/workshop_survey/post/:enrollment_code` and just returns `new_post_foorm` anyways so I just made `new_post` the 'get survey by enrollment code' action in the controller. This way, whenever a survey is obtained by the enrollment code it always uses the same method (and within that, it determines whether to use foorm or not).

File: dashboard/app/controllers/pd/workshop_daily_survey_controller.rb (hunk 2 above)
Comment (TurnerRiley) [reply dashboard/app/controllers/pd/workshop_daily_survey_controller.rb:75]: It was previously a controller action but I updated it to be a helper method the same way I did with `new_post` vs. `new_post_foorm`

File: dashboard/app/controllers/pd/workshop_daily_survey_controller.rb (hunk 3 above)
Comment (TurnerRiley) [reply dashboard/app/controllers/pd/workshop_daily_survey_controller.rb]: `new_post_foorm` ends up returning a `render` for the `foorm` so I figured returning a `url` here wasn't going to work and I'd have to either do a render or redirect. I did so much tinkering I can't remember if I tested just a `return` for the url so let me try that super quick!

Comment (TurnerRiley) [reply dashboard/app/controllers/pd/workshop_daily_survey_controller.rb]: Okay yep just double checked it's expecting a template so just a url doesn't work, it needs a `render` or `redirect`

File: dashboard/app/controllers/pd/workshop_daily_survey_controller.rb [hunk 4]
```diff
@@ -64,16 +64,14 @@ def new_pre_foorm
     end
 
     # General post-workshop survey using foorm system.
-    # GET '/pd/workshop_post_survey?enrollmentCode=code'
     # Enrollment code is an optional parameter, otherwise will show most recent workshop.
     #
     # If the post-survey has been already completed, will redirect to thanks page.
     def new_post_foorm
       new_general_foorm(survey_names: POST_SURVEY_CONFIG_PATHS, day: nil)
```
Comment (bethanyaconnor) [inline dashboard/app/controllers/pd/workshop_daily_survey_controller.rb]: maybe a follow-up but now this method just wraps another method. Is it worth just deleting this method?

**Review by bethanyaconnor — APPROVED** [review]
Looks good!

File: dashboard/app/controllers/pd/workshop_daily_survey_controller.rb (hunk 4 above)
Comment (TurnerRiley) [reply dashboard/app/controllers/pd/workshop_daily_survey_controller.rb]: Oh true! I'm already making the update to fix the quotes so I'll fix this too!

**Review by bethanyaconnor — APPROVED** [review]
🎉

**Pull request merged**